test/testutil/main.c

```diff
@@ -8,13 +8,86 @@
  */
 
 #include "../testutil.h"
+#include "../../e_os.h"
+#include "output.h"
+
+#include <string.h>
+
+static size_t arg_count;
+static char **args;
+static unsigned char arg_used[1000];
+
+static void check_arg_usage(void)
+{
+    size_t i;
+
+    for (i = 0; i < arg_count; i++)
+        if (!arg_used[i+1])
+            test_printf_stderr("Warning ignored command-line argument %d: %s\n",
+                               i, args[i+1]);
+}
 
 int main(int argc, char *argv[])
 {
-    int ret;
-    setup_test();
+    int ret = EXIT_FAILURE;
+
+    arg_count = argc - 1;
+    args = argv;
+
+    setup_test_framework();
+
+    if (setup_tests())
+        ret = run_tests(argv[0]);
+    cleanup_tests();
+    check_arg_usage();
+
+    return pulldown_test_framework(ret);
+}
+
+const char *test_get_program_name(void)
+{
+    return args[0];
+}
 
-    ret = test_main(argc, argv);
+char *test_get_argument(size_t n)
+{
+    if (n > arg_count)
+        return NULL;
+    if (n + 1 < OSSL_NELEM(arg_used))
+        arg_used[n + 1] = 1;
+    return args[n + 1];
+}
 
-    return finish_test(ret);
+size_t test_get_argument_count(void)
+{
+    return arg_count;
 }
+
+int test_has_argument(const char *option)
```

Comment:
To get away from the similarity with `test_get_argument`, you could rename this to `test_has_option`